src/OrleansProviders/Streams/Memory/MemoryStreamProvider.cs

```diff
@@ -0,0 +1,12 @@
+﻿using Orleans.Providers.Streams.Common;
+
+namespace Orleans.Providers.Streams.Memory
+{
+    /// <summary>
+    /// This is a persistent stream provider that uses in-memory grain to queue the events.
+    /// This is primarily for test purposes.
```

Comment:
This provider is mostly modeled after EventHub.  One of the reasons for introducing it was to allow developers to develop against a persistent recoverable stream provider, like EH, without having to have a Hub per developer.  It is not at all unreasonable to have an in-memory mode supported by each persistent stream provider rather than a separate in-memory provider.  The original EventHub provider worked much like you describe, it could write to EventHub partitions or in-memory queues.  That pattern turned out to be more difficult than expected.  The current EventHub stream provider is the most complex provider we have so we're trying not to add any more complexity to it until we complete it's feature set.

Once the EventHub stream provider is complete and more stable, I'd be very open to moving to that model, but to avoid introducing further complexity into the EventHub stream provider we decided to introduce this capability via a new provider.